src/state.cpp

```diff
@@ -580,6 +580,21 @@ State::do_export(const std::string& label,
   return _suite.expand_with_label(secret, "exporter", context, size);
 }
 
+std::vector<Credential>
+State::get_leaf_credentials() const
+{
+  std::vector<Credential> creds;
```

Comment:
You might get a little efficiency by allocating this to full size (`_tree.size()`) and then truncating it at the end (`creds.resize(leaf_count)`, where `leaf_count` is the number of non-blank leaves you found)